velox/functions/lib/window/NthValueBase.h

```diff
@@ -128,12 +114,12 @@ class NthValueFunction : public exec::WindowFunction {
     offsets_->resize(numRows);
     partition_->extractColumn(
         offsetIndex_, partitionOffset_, numRows, 0, offsets_);
-
+    auto offsets = offsets_->asFlatVector<int64_t>();
```

Comment:
This presumes that offsets is an int64 vector. Please check the type before initializing this.

Reply:
I moved this to Presto NthValue class also, thanks

Reply:
Your current re-factoring moves a lot of the logic. I think the older code was better. It was sufficient to just check the type before type casting it (and throw an error if not int64).